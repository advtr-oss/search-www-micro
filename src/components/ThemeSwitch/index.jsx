import React from 'react'
import PropTypes from 'prop-types'
import { ThemeSwitch as ADVThemeSwitch } from '@advtr/tidy'

export default class ThemeSwitch extends React.Component {
  static propTypes = {
    onClick: PropTypes.func
  }

  render () {
    return <ADVThemeSwitch onClick={this.props.onClick} aria-label='Toggle Dark Mode' />
  }
}
