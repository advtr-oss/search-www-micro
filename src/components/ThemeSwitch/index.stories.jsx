import React from 'react'

import { action } from '@storybook/addon-actions'

import ADVThemeSwitch from './index'

export default {
  title: 'Switch/Theme',
  component: ADVThemeSwitch
}

export const ThemeSwitch = () => <ADVThemeSwitch onClick={action('change-theme')} />
