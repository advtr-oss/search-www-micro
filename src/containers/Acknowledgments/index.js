import React, { useEffect, useState, memo } from 'react'
import PropTypes from 'prop-types'
import { FormattedMessage } from 'react-intl'
import { withRouter } from 'react-router-dom'

import { DateTime } from 'luxon'

import messages from './messages'

import Wrapper from './Wrapper'

import Header from '../../components/Header'

// TODO: Add more to his, i.e links to github's etc
const Acknowledgment = ({ awk }) => (
  <div>
    <Header level='h3' title={`${awk.name} @ ${awk.version}`} />
    {awk.description && (
      <span className='description'>
        {awk.description}
      </span>
    )}
    <div className='separator'>---------------------------------------------------</div>
  </div>
)

Acknowledgment.propTypes = {
  awk: PropTypes.shape({
    name: PropTypes.string,
    version: PropTypes.string,
    description: PropTypes.string
  })
}

const AcknowledgmentsPage = ({ history }) => {
  const [acknowledgments, setAcknowledgments] = useState(null)

  // Have this in an useEffect with a loader and Memo'd
  useEffect(() => {
    if (!acknowledgments) {
      setAcknowledgments(require('../../assets/acknowledgments.json'))
    }
  }, [acknowledgments])

  return (
    <Wrapper className='container-lg'>
      <div className='header'>
        <div className='meta'>
          <FormattedMessage {...messages.header}>
            {(title) =>
              (<Header level='h1' title={title[0]} />)}
          </FormattedMessage>
          {acknowledgments && (
            <FormattedMessage {...messages.lastUpdated}>
              {(lastUpdated) =>
                (<span>{lastUpdated[0]}: {DateTime.fromMillis(acknowledgments.date).toLocaleString(DateTime.DATE_FULL)}</span>)}
            </FormattedMessage>
          )}
        </div>
        <div className='home'>
          <button className='bck-btn' onClick={(event) => { event.preventDefault(); history.push('/') }}>
            Home
          </button>
        </div>
      </div>
      {acknowledgments && (
        <div className='codeblock'>
          {Object.keys(acknowledgments.dependencies).map((key) => {
            const dep = acknowledgments.dependencies[key]
            return (
              <Acknowledgment key={key} awk={dep} />
            )
          })}
        </div>
      )}
    </Wrapper>
  )
}

AcknowledgmentsPage.propTypes = {
  history: PropTypes.shape({
    push: PropTypes.func
  })
}

export default withRouter(memo(AcknowledgmentsPage))
