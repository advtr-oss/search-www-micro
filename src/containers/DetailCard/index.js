/**
 * The main part of the Detail tree, this is here to determine if we're on mobile or anything else to
 * display the correct view, should handle any analytics while it's here too
 * */

import React, { useEffect } from 'react'
import { useMediaQuery } from 'react-responsive'

import { DetailView, DetailModalView } from '../DetailView'

import useTheme from '../../hooks/useTheme'
import useTracking from '../../hooks/useTracking'
import { ADVTR_DETAIL_EVENT, ADVTR_DETAIL_DISPLAYED } from '../../utils/analyticsConstants'

const breakpointToNumber = (bp) => parseInt(bp.replace('px', ''))

const DetailCard = () => {
  const { trackGoogleAnalyticsEvent } = useTracking()
  const { main: { sizes: { breakpoints } } } = useTheme()

  const isMobile = useMediaQuery({ maxWidth: breakpointToNumber(breakpoints.sm) })

  useEffect(() => {
    trackGoogleAnalyticsEvent('event', ADVTR_DETAIL_DISPLAYED, {
      category: ADVTR_DETAIL_EVENT,
      label: 'detail',
      value: window.innerWidth
    })
  })

  return isMobile ? <DetailModalView /> : <DetailView />
}

export default DetailCard
