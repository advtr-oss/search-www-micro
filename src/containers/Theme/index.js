import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { createSelector } from 'reselect'
import { ThemeProvider } from 'styled-components'

import { makeSelectDarkMode } from './selectors'

import theme from '../../themes'

const Theme = ({ darkMode, children }) => {
  return <ThemeProvider theme={{ ...theme, mode: darkMode }}>{children}</ThemeProvider>
}

Theme.propTypes = {
  darkMode: PropTypes.bool,
  children: PropTypes.node.isRequired
}

const mapStateToProps = createSelector(
  makeSelectDarkMode(),
  darkMode => ({
    darkMode
  })
)

export default connect(mapStateToProps)(Theme)
