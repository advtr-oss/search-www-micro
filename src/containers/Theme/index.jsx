/**
 * This is just to keep an eye on this, and we have one point of call for it
 * */

export { ThemeProvider as Theme } from '@advtr/tidy'
