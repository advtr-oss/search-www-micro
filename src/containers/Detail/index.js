/**
 * Grouping of the cards, may not be important but will be okay for now
 * */

import React, { } from 'react'
// import PropTypes from 'prop-types'

import SearchCard from '../SearchCard'
import DetailCard from '../DetailCard'
import DetailCardModal from '../DetailCardModal'

import useDeviceState from '../../hooks/useDevice'

const Detail = () => {
  const { isMobile } = useDeviceState()

  return (
    <>
      <SearchCard />
      {isMobile ? <DetailCardModal /> : <DetailCard />}
    </>
  )
}

Detail.propTypes = {

}

export default Detail
