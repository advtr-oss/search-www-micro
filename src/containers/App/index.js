import React, { useEffect } from 'react'
import { Switch, Route } from 'react-router-dom'

import ReactGA from 'react-ga'
import { nanoid } from 'nanoid'

import Theme from '../Theme'

import TrackingProvider from '../TrackingProvider'
import GlobalStyle from '../../components/GlobalStyle'

import SearchPage from '../SearchPage'
import AcknowledgmentsPage from '../Acknowledgments'

const set = (obj) => {
  window.ga && ReactGA.set(obj)
}

export default function App () {
  // Set a session token, for our logging purposes
  useEffect(() => {
    const session = nanoid()
    if (window && !window.sessionStorage.getItem('sessionToken')) {
      window.sessionStorage.setItem('sessionToken', session)
      set({ dimension1: session })
    } else if (window.sessionStorage.getItem('sessionToken')) {
      set({ dimension1: window.sessionStorage.getItem('sessionToken') })
    }
  })

  return (
    <Theme>
      <TrackingProvider>
        <GlobalStyle />
        <Switch>
          <Route path='/acknowledgments' component={AcknowledgmentsPage} />
          <Route path='*' component={SearchPage} />
        </Switch>
      </TrackingProvider>
    </Theme>
  )
}
