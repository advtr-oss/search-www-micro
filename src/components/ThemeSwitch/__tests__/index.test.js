/**
 * @jest-environment jsdom
 */

import React from 'react'
import { render, screen, fireEvent } from '@testing-library/react'

import ThemeSwitch from "../index";

import { Theme } from "../../../containers/Theme";

const theme = (ui, opts) => render((
  <Theme>
    <>{ui}</>
  </Theme>
), opts)

describe('ThemeSwitch', function () {
  it('should render switch', function () {
    const { container } = theme(<ThemeSwitch />)
    expect(container.firstChild).toMatchSnapshot()
  });

  it('should have default theme value', function () {
    theme(<ThemeSwitch />)

    const checkbox = screen.getByLabelText('theme-toggle')
    expect(checkbox.checked).toBeFalsy()
  });

  it('should fire click event', function () {
    const { container } = theme(<ThemeSwitch />)
    fireEvent.click(screen.getByLabelText('theme-toggle'))

    expect(container.firstChild).toMatchSnapshot()
  });
});
