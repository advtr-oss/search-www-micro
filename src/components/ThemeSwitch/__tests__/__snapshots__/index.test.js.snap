// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeSwitch should fire click event 1`] = `
.c0 {
  box-sizing: content-box;
}

.c0 > * {
  box-sizing: content-box !important;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  width: 200px;
  height: 100px;
  border-radius: 100px;
  border: 5px solid rgba(205,217,229,0.1);
  background: #2d333b;
  -webkit-transition: all 350ms ease-in;
  transition: all 350ms ease-in;
}

.c3::before {
  -webkit-animation-name: jbTuch;
  animation-name: jbTuch;
  -webkit-animation-duration: 350ms;
  animation-duration: 350ms;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  -webkit-transition: all 350ms ease-in;
  transition: all 350ms ease-in;
  content: "";
  width: 82px;
  height: 82px;
  border: 5px solid #eac55f;
  top: 4px;
  left: 4px;
  position: absolute;
  border-radius: 82px;
  background: #daaa3f;
}

.c3::after {
  -webkit-transition-delay: 0ms;
  transition-delay: 0ms;
  -webkit-transition: all 250ms ease-in;
  transition: all 250ms ease-in;
  position: absolute;
  content: "";
  left: 143px;
  top: 23px;
  width: 10px;
  height: 10px;
  background: transparent;
  border-radius: 50%;
  opacity: 0;
}

.c4:checked +.c2::before {
  background: #ffffff;
  border-color: rgba(208,214,220,0.5);
  -webkit-animation-name: coksJc;
  animation-name: coksJc;
  -webkit-animation-duration: 350ms;
  animation-duration: 350ms;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
}

.c4:checked +.c2::after {
  -webkit-transition-delay: 350ms;
  transition-delay: 350ms;
  opacity: 1;
  box-shadow: rgba(208,214,220,0.5) -13px 0 0 2px,rgba(208,214,220,0.5) -24px 14px 0 -2px;
}

.c1 {
  display: none;
}

.c1:checked +.c2::before {
  background: #cdd9e5;
  border-color: rgba(55,62,71,0.5);
  -webkit-animation-name: coksJc;
  animation-name: coksJc;
  -webkit-animation-duration: 350ms;
  animation-duration: 350ms;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
}

.c1:checked +.c2::after {
  -webkit-transition-delay: 350ms;
  transition-delay: 350ms;
  opacity: 1;
  box-shadow: rgba(55,62,71,0.5) -13px 0 0 2px,rgba(55,62,71,0.5) -24px 14px 0 -2px;
}

<div
  class="c0"
>
  <input
    aria-label="theme-toggle"
    class="c1"
    id="theme-toggle"
    type="checkbox"
  />
  <label
    class="c2 c3"
    for="theme-toggle"
  />
</div>
`;

exports[`ThemeSwitch should render switch 1`] = `
.c0 {
  box-sizing: content-box;
}

.c0 > * {
  box-sizing: content-box !important;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  width: 200px;
  height: 100px;
  border-radius: 100px;
  border: 5px solid rgba(27,31,36,0.15);
  background: #ffffff;
  -webkit-transition: all 350ms ease-in;
  transition: all 350ms ease-in;
}

.c3::before {
  -webkit-animation-name: jbTuch;
  animation-name: jbTuch;
  -webkit-animation-duration: 350ms;
  animation-duration: 350ms;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  -webkit-transition: all 350ms ease-in;
  transition: all 350ms ease-in;
  content: "";
  width: 82px;
  height: 82px;
  border: 5px solid #fae17d;
  top: 4px;
  left: 4px;
  position: absolute;
  border-radius: 82px;
  background: #eac54f;
}

.c3::after {
  -webkit-transition-delay: 0ms;
  transition-delay: 0ms;
  -webkit-transition: all 250ms ease-in;
  transition: all 250ms ease-in;
  position: absolute;
  content: "";
  left: 143px;
  top: 23px;
  width: 10px;
  height: 10px;
  background: transparent;
  border-radius: 50%;
  opacity: 0;
}

.c1 {
  display: none;
}

.c1:checked +.c2::before {
  background: #ffffff;
  border-color: rgba(208,214,220,0.5);
  -webkit-animation-name: coksJc;
  animation-name: coksJc;
  -webkit-animation-duration: 350ms;
  animation-duration: 350ms;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
}

.c1:checked +.c2::after {
  -webkit-transition-delay: 350ms;
  transition-delay: 350ms;
  opacity: 1;
  box-shadow: rgba(208,214,220,0.5) -13px 0 0 2px,rgba(208,214,220,0.5) -24px 14px 0 -2px;
}

<div
  class="c0"
>
  <input
    aria-label="theme-toggle"
    class="c1"
    id="theme-toggle"
    type="checkbox"
  />
  <label
    class="c2 c3"
    for="theme-toggle"
  />
</div>
`;
